api/v1/helpers/verbs/doPut.js

```diff
@@ -29,11 +30,21 @@ const logAPI = require('../../../../utils/loggingUtil').logAPI;
  *  resource type to put.
  */
 function doPut(req, res, next, props) {
+  let modelInst;
   const toPut = req.swagger.params.queryBody.value;
   const puttableFields =
     req.swagger.params.queryBody.schema.schema.properties;
   u.findByKey(props, req.swagger.params)
   .then((o) => {
+    modelInst = o;
+    return featureToggles.isFeatureEnabled('enforceWritePermission') ?
+    u.isWritable(req, o) : true;
+  })
+  .then((ok) => {
+    if (!ok) {
+      return u.forbidden(next, props.modelName);
+    }
+
```

Comment:
I see that this code is repeated at many places. Is it possible to create and use a common function in util which returns a promise with resolve(modelInst) and reject(forbiddenErr).